### ISSUE TITLE
Possible to have line info in errors ?

### ISSUE THREAD

**Issue by StyMaar** [gura-conf/serde-gura#5]
Every time I make a mistake in my gura file, I get an error and I need to figure out where this damn error is located. Having a line information would be a massive quality of life improvement!

So far I've only met one instance of error coming with the line number 

> Err: Input text does not have a valid Gura format. Parsing failed with error \"Expected end of string but got 'a' at line 65 position 1800\", in file Some("my_file_path.ura")

But all other errors have been a hide and seek game between me and my configuration file.

**Comment by Genarito** [comment]
Hi! That's true, in fact, the parser that Serde Gura uses in the back throws the string position where the parse failed, so it shouldn't be a big change to add the enhancement to the library. I will be applying the change and upload the new version by tonight or tomorrow at the latest :D

**Comment by StyMaar** [comment]
Great thank you a lot!

Just so you know, only half of the errors I'm facing are related to parsing (the wrong number of white spaces, or stealthy tabs, mostly), but the other half is deserialization-related (like: missing field, often because of indentation mistakes).

**Comment by StyMaar** [comment]
Having line numbers for parse errors would be nice, of course, but having both would be even nicer ^^.

**Comment by Genarito** [comment]
You will have both! 💯

**Comment by Genarito** [comment]
Hi @StyMaar. I'm testing right now some parsing errors and I'm getting the errors message in Serde Gura. But I realized that the line and position where the error is detected is the wrong one. For example, the following Gura block:

```yaml
tango_singers: [
    test: "Some string", H   # <---- This "H" is a parsing error
]
```

Throws an error `Expected end of string but got 't'` in position 0 and line 0 because it's getting the parent block `tango_singers` as a reference. I must update the parser before I can fix this issue on Serde Gura. It'll take more than today as I'm a little busy, but I promise I'll fix it as soon as possible, thanks for the patience!

**Comment by Genarito** [comment]
I am pleased to announce that I have now released version `0.1.4` of the [crate](https://crates.io/crates/serde_gura) :partying_face: I must thank you for reporting this problem as I found several bugs in the parser bug report. Now Serde Gura, and [Rust](https://github.com/gura-conf/gura-rs-parser), [Python](https://github.com/gura-conf/gura-python-parser) (and soon [JS](https://github.com/gura-conf/gura-js-parser)) parsers have:

- Line and global position of the error that occurred.
- Now all errors report position (previously only `ParseError` did).
- Improved some messages to help users to solve the problem

In addition, I took advantage of the work on this tool to also publish the new version with a fix for [a bug reported recently](https://github.com/gura-conf/gura/issues/15) that prevented the use of numeric keys during deserialization. I will close the issue, feel free to reopen it if you consider that the problem is still not solved :+1: 

Thanks for using and making Gura better! :smile:

**Comment by StyMaar** [comment]
Thanks a lot for your work !
